android/src/main/java/app/notifee/core/Notifee.java

```diff
@@ -119,6 +120,17 @@ public void cancelAllNotificationsWithIds(
             });
   }
 
+  @KeepForSdk
+  public void canCreateTimeStampTrigger(MethodCallResult<Boolean> result) {
```

Comment:
@vincent-paing if we extend `getNotificaationSettings`, would we still need this API?

Reply:
I think we can keep both and mark this as deprecated to be removed in future in that case? Or do you feel like we should get it right the first time to prevent long tail?

Reply:
I think these are all new APIs, no need to follow a deprecation cycle, they have never been released so technically did not exist and cannot be deprecated

Reply:
@mikehardy For now, we would not need to but would we need to mark this as deprecate if we put this boolean flag in `getNotificationSettings` in future?

Reply:
Hmm - I guess it comes down to how strongly @helenaford feels about mixing the concerns whether it is necessary to even keep a separate API.

To define terms just a bit since we are all asynchronous and I may not communicate clearly otherwise: by mixing concerns I mean that permission to set alarms is not really permission to send notifications, they are separate technically. 

Now, does that matter, to me?
From a notification module perspective the only use for the alarm is to display a notification, so from that perspective, I think they are close enough to the same thing that I'm okay mixing them. As a separate criteria for whether it's okay or not I think about the developer use case, and I can't imagine a case where I wouldn't want to run all the permissions / capability checks at once, so to me, I think mixing makes sense in this library and for our developers.

So I like the idea of putting the key in notification settings.

At that point, it feels having a separate API does not have much value. So I personally would put the key in notification settings (and have it come back when the API for getting settings is called), and I'd get rid of the separate API

Helena may or may not feel differently :-). And either way, I really want to emphasize that this is in my mind a tiny bit of polish on a PR that's fantastic, and I appreciate the PR a lot, excited to see it merged. Thanks

Reply:
I can tell that all of us felt that putting that in notification settings might be more useful for developers than to make it a separate API. In architectural perspective, I will put the native API boolean check into `AlarmUtils` so that it's still logically separated in native side, delete the separate API on the react native part of the lib and merged it into `getNotificationSettings`. This way, we can prevent long tail issue of having to maintain an extra function, and also wouldn't require us to release a breaking change in future when we remove this API completely. 

I might be able to find some free time during weekends and will make changes then

Reply:
thanks @vincent-paing. sounds good. As soon as that's done, I'll hit merge and get it released 😎